### PR TITLE
Add support for element iteration in IndexableScanner

### PR DESCRIPTION
**Pull request by youngar**
cc @rwy0717 @amicic @dmitripivkine

### CODE REVIEW

**Comment by youngar** [comment]
@genie-omr build all

File: gc/base/IndexableObjectScanner.hpp [hunk 1]
```diff
@@ -79,19 +107,24 @@ class GC_IndexableObjectScanner : public GC_ObjectScanner
...
-	MMINLINE void
-	initialize(MM_EnvironmentBase *env)
+	MMINLINE fomrobject_t *getScanPtr() { return _scanPtr; }
+
+	/**
+	 * Get the address of the next element in the array.
+	 * @return NULL if there are no more elements
+	 */
+	MMINLINE fomrobject_t *
+	next()
```
Comment (amicic) [inline gc/base/IndexableObjectScanner.hpp]: I believe you said you'll rename this to nextIndexableElement

Comment (youngar) [reply gc/base/IndexableObjectScanner.hpp]: I thought I had. I think I'm missing other changes too

**Comment by youngar** [comment]
Missed some changes.  Not sure how I messed up my squash.

**Comment by rwy0717** [comment]
@genie-omr build all

**Comment by AdamBrousseau** [comment]
genie-omr build zos

**Comment by youngar** [comment]
As far as I can tell, AIX build failed because I was not returning a value after `assert_mm_unreachable`.  It did not really report a clear error.
@genie-omr build all

**Comment by rwy0717** [comment]
@genie-omr build zos

**Comment by youngar** [comment]
Deleted an extra newline.  All builds except z/os passed (infrastructure issue).

**Pull request merged**